### ISSUE TITLE
Search no longer works with Coq 8.12

### ISSUE THREAD

**Issue by silene** [coq/coq#12647]
In Coq 8.12, `Search` no longer returns any result. This might be a duplicate of #12525.

```coq
From Coq Require Lia.

Module Type Foo.
Axiom P : nat -> Prop.
Axiom L : P 0.
End Foo.

Module Bar (F : Foo).
Search F.P. (* empty, except for a warning about SSReflect, which I did not even require indirectly *)
```

This is a regression with respect to Coq 8.11.

If I remove `Require Lia`, then it suddenly works. Unfortunately, `Lia` is most certainly going to be directly or indirectly required by any user development.

**Comment by gares** [comment]
I guess this is then 2 bugs. One is likely to be a duplicate of #12525 and the other is a spurious warning.

**Comment by silene** [comment]
The thing is, I am not even completely convinced that it is a duplicate of #12525. Indeed, an incorrectly empty delta resolver should work just fine, unless I am missing something. Indeed, `F.L` is not coming from the external world. Also, I do not understand why the `Require` statement can have any effect (changing `Lia` to something else, e.g., `ZArith`, makes the example work).

**Comment by gares** [comment]
Crazy guess, a broken blacklist based on module names? `L` = `Lia` != `ZArith`?

**Comment by ejgallego** [comment]
Adding a bit of printf in Search may reveal the problem quickly.

**Comment by ejgallego** [comment]
Note that the problems with `Search` seem serious as to consider them release critical. so if anyone has a spare cycle to look into the two bugs would be great, otherwise the changes may have to be delayed to 8.13 [@Zimmi48 and I didn't discuss about this yet, but on my table I'm certainly worried about the bug]

**Comment by SkySkimmer** [comment]
Just  `Require Vector.` is sufficient to see the bug, but inlining the contents of Vector.v does not trigger the bug.

**Comment by herbelin** [comment]
As already checked by @SkySkimmer, this is fixed by #12537.

> The thing is, I am not even completely convinced that it is a duplicate of #12525. Indeed, an incorrectly empty delta resolver should work just fine, unless I am missing something, as `F.L` is not coming from the external world. 

The problem is not with `F.L`, the problem is with the other definitions in the search space. The bug was raising a `Not_found` which was interrupting the whole search (see block `try Declaremods.iter_all_segments iter_obj with Not_found -> ()` in `search.ml`).

> Also, I do not understand why the `Require` statement can have any effect (changing `Lia` to something else, e.g., `ZArith`, makes the example work).

Presumably, any `Require` which loads a library containing an `Include` is trigerring the problem.

**Comment by Zimmi48** [comment]
> As already checked by @SkySkimmer, this is fixed by #12537.

What about the part regarding the spurious warning message? Any idea about that?

**Comment by herbelin** [comment]
I don't see this warning with #12537 (but I did not test with current master).

**Comment by SkySkimmer** [comment]
Couldn't reproduce the warning, maybe @silene loads ssr in his rcfile?

**Comment by silene** [comment]
The warning is a bit strange. The testcase as I wrote it does not trigger it. But if I add and *remove* the line `From Coq Require ssreflect`, then the warning triggers. (In other words, I got the warning because I minimized a testcase that was originally using SSReflect indirectly.) So, it seems like the warning is controlled by some global state that is not properly updated when changing the required modules. That is a minor issue.
